plugins/templates/create.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
```

Comment:
> For example, scmUri is a unique field of a pipeline.
`scmuri` is not a consistent form. It can be `https` or `git` protocols. 

> There cannot be any two jobs with the same pipelineId + name

The initial design for the job was to have it canonical and use a `/pipelines/{id}/jobs/{job}`. We then switched it to `/pipelines/{id}/jobs/{id}`. Do you recall the problems we had with it?
 
Templates with different names are unique. Just because you have an `id` associated with all the other URIs, it doesn't mean this one has to also. Github repositories have an associated `id` yet that is not exposed to the user. 

We've already solved this problem so I'm confused why this is a contention point. What makes this so unique that it can't follow the same pattern?

@stjohnjohnson even echoes a similar concern here: https://github.com/screwdriver-cd/config-parser/pull/32#discussion_r105335166 A POST is never cached anyways, but the endpoint structure is still applicable here.